firmware/library/L1_Drivers/gpio.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+class GpioInterface
+{
+ public:
+    enum PinDirection : uint8_t
+    {
+        kInput  = 0,
+        kOutput = 1
+    };
+    enum PinOutput : uint8_t
+    {
+        kLow  = 0,
+        kHigh = 1
+    };
+    virtual void SetAsInput(void)                             = 0;
+    virtual void SetAsOutput(void)                            = 0;
+    virtual void SetDirection(
```

Comment:
Don't have a default of this. It doesn't read right if the user doesn't supply a parameter. Keep the enum though.